### DISCUSSION TITLE
`Undefined` return from useReadContract

### DISCUSSION THREAD

**Discussion by AbhinavMir** [wevm/wagmi#3687] [Q&A]
```
const results = useReadContract({
    address: "0x10e2161f53B7F00Cab5123B280B0e622c61B4Aba",
    abi: abi,
    functionName: "getVaults",
    chainId: bsc.id,
  });

  console.log("vaults", results);
```

When I do this, I get an `undefined` error. Tried to do the `config` setting as well, still undefined. Sepolia also gives me an undefined return. Tried it with Python's web3js, works flawlessly. Where am I going wrong? (Both examples use the same ABI, the one in wagmi is just an exported TS object).

```
from web3 import Web3
import json

// swtiched from BSC to Sepolia because I thought it might be an outdated RPC endpoint
w3 = Web3(Web3.HTTPProvider('https://ethereum-sepolia-rpc.publicnode.com'))

contract_address = '0x10e2161f53B7F00Cab5123B280B0e622c61B4Aba'
with open('abi.json', 'r') as f:
    abi = json.load(f)

contract = w3.eth.contract(address=contract_address, abi=abi)

result = contract.functions.getVaults().call()
print(result)
```

**Comment by vrypan** [comment]
I'm having the same problem, with an erc721 contract. I'm just learning react+wagmi and this is driving me crazy.

Other hooks, like `useBlockNumber()` and `useAccount()` work perfectly fine, inside the same component. But I can't get a simple `totalSupply` from an ERC721.

Can someone please help?

**Reply by tmm** [reply]
What version of Wagmi, Viem, and TypeScript are you using? In the future, please provide a minimal reproduction so I can look into it instead of having to ask.

**Reply by vrypan** [reply]
TBH, I have no idea. I created a new project using rainbow-kit's instructions and now it works.

However, I noticed something that could be one of the reasons of my problems yesterday: Rendering a uint256 value, will show nothing, unless I convert it to string. This was not obvious to me (admittedly, a react noob), and I may have gone down the wrong path thinking that no value was returned.
